common/lib/dependabot/file_fetchers/base.rb

```diff
@@ -861,19 +861,35 @@ def decode_binary_string(str)
 
       sig { params(path: String).returns(T::Array[String]) }
       def find_submodules(path)
-        SharedHelpers.run_shell_command(
-          <<~CMD
-            git -C #{path} ls-files --stage
-          CMD
-        ).split("\n").filter_map do |line|
+        lfsEnabled = true
+        commandString = getCommandString(path,lfsEnabled)
+        /debugger
+        p commandString/
+        SharedHelpers.run_shell_command(commandString
+         ).split("\n").filter_map do |line|
           info = line.split
 
           type = info.first
           path = T.must(info.last)
-
+debugger
           next path if type == DependencyFile::Mode::SUBMODULE
         end
       end
+
+      def getCommandString(path,lfsEnabled)
+	      #the HEREDOC command will see any stray spaces.
+        return "git -C #{path} ls-files --stage" unless lfsEnabled
+        Dependabot.logger.warn("LFS is enabled in this repo.  Please use an LFS enabled client")
+        commandString = "CWD=\"#{__dir__}\";cd #{path};git-lfs ls-files --stage;cd $CWD"
```

Comment:
I suggested something a little simpler above, but just adding a couple notes:
* [`__dir__`](https://docs.ruby-lang.org/en/3.3/Kernel.html#method-i-__dir__) is the name of the directory that contains this source code file, not the current working directory of the process
* Rather than setting an environment variable for the directory, this is a good use case for [`Dir.chdir`](https://docs.ruby-lang.org/en/3.3/Dir.html#method-c-chdir). This method changes into the given directory, runs whatever is inside block, and then changes back to the original directory.